fs/accounting.go

```diff
@@ -21,20 +21,70 @@ var (
 	tokenBucketMu   sync.Mutex // protects the token bucket variables
 	tokenBucket     *tb.Bucket
 	prevTokenBucket = tokenBucket
+	currLimitMu     sync.Mutex // protects changes to the timeslot
+	currLimit       BwTimeSlot
 )
 
 // Start the token bucket if necessary
 func startTokenBucket() {
-	if bwLimit > 0 {
-		tokenBucket = tb.NewBucket(int64(bwLimit), 100*time.Millisecond)
-		Log(nil, "Starting bandwidth limiter at %vBytes/s", &bwLimit)
+	currLimitMu.Lock()
+	currLimit, err := bwLimit.LimitAt(time.Now())
+	currLimitMu.Unlock()
+	// This should never happen! It means we got an invalid time from time.Now()
+	if err != nil {
+		Log(nil, "This error should not happen. Please open a bug: %v", err)
```

Comment:
This should be `ErrorLog` so that it is visible to users even with `-q`

However below I suggest that LimitAt shouldn't return an error at all...